handlers/user-password-reset-url.js

```diff
@@ -0,0 +1,139 @@
+    *                       returns { error:{null || {new Error("Error Message")} }, value: {normalize(value)}}
+    * }
+    */
+   inputValidation: {
+      email: { string: { email: { allowUnicode: true } }, required: true },
+   },
+
+   /**
+    * fn
+    * our Request handler.
+    * @param {obj} req
+    *        the request object sent by the
+    *        api_sails/api/controllers/user_manager/user-password-reset-request.
+    * @param {fn} cb
+    *        a node style callback(err, results) to send data when job is finished
+    */
+   fn: function handler(req, cb) {
+      req.log("user_manager.user-password-reset-url:");
+
+      // get the AB for the current tenant
+      ABBootstrap.init(req)
+          .then(async (AB) => { // eslint-disable-line
+
+            req.log(req.param("email"));
+
+            // 1) get User
+            const cond = { email: req.param("email") };
+            const User = AB.objectUser();
+            const list = await req.retry(() =>
+               User.model().find({ where: cond, populate: false })
+            );
+
+            if (!list || !list[0]) {
+               // Q: is there any additional management in this case?
+               // eg: do we mark how many failed attempts and then block that browser?
+               req.log("User not found: ", JSON.stringify(cond));
+
+               const error = new Error("E_NOT_FOUND");
+
+               cb(error);
+               return;
+            }
+            const user = list[0];
+            req.log(`User Requesting Password Reset Link: ${user.username}`);
+
+            // 2) create new authToken
+            //   - authToken table needs to have authToken, user, metadata (for storing route/app specific info)
+
+            // https://security.stackexchange.com/questions/94630/token-based-authentication-whats-a-good-token-length
+            const token = await nanoid(16);
+
+            // store our current request context:
+            const context = {
+               username: user.username,
+               email: user.email,
+               tenantID: req.tenantID(),
+            };
+
+            // expires in 10 minutes
+            // const expires = new Date(Date.now() + numMinutes*60000);
+            const expires = new Date(Date.now() + 600000); // <- 10 min
+
+            const SiteToken = AB.objectToken();
+            await req.retry(() =>
+               SiteToken.model().create({ token, context, expires })
+            );
+            req.log("newToken:", token);
+
+            // 3) generate email with Auth Link
+            req.serviceRequest(
+               "tenant_manager.tenant-url",
+               {
+                  uuid: req.tenantID(),
+               },
+               (err, results) => {
+                  if (err) {
+                     req.notify.developer(err, {
+                        context:
+                           "Service:tenant_manager.tenant-url: Error initializing ABFactory",
```

Comment:
OK, so this isn't an "Error initializing ABFactory".  it is an `Error requesting tenant-url.`